src/main/java/org/testng/JarFileUtils.java

```diff
@@ -52,26 +59,59 @@ private boolean testngXmlExistsInJar(File jarFile, List<String> classes) throws
         boolean foundTestngXml = false;
         try (JarFile jf = new JarFile(jarFile)) {
             Enumeration<JarEntry> entries = jf.entries();
+            File file = createTempDir();
+            String suitePath = null;
             while (entries.hasMoreElements()) {
                 JarEntry je = entries.nextElement();
-                if (matchesXmlPathInJar(je)) {
-                    Collection<XmlSuite> parsedSuites = Parser.parse(jf.getInputStream(je), processor);
-                    for (XmlSuite suite : parsedSuites) {
-                        // If test names were specified, only run these test names
-                        XmlSuite suiteToAdd = suite;
-                        if (testNames != null) {
-                            suiteToAdd = XmlSuiteUtils.cloneIfContainsTestsWithNamesMatchingAny(suite, testNames);
-                        }
-                        suites.add(suiteToAdd);
+                if (je.getName().toLowerCase().endsWith(".xml")){
+                    String jeName = je.getName();
+                    InputStream  inputStream = jf.getInputStream(je);
+                    File copyFile = new File(file, jeName);
+                    Files.copyFile(inputStream, copyFile);
+                    if (matchesXmlPathInJar(je)) {
+                        suitePath = copyFile.toString();
                     }
-                    return true;
-                } else if (isJavaClass(je)) {
+                }else if (isJavaClass(je)) {
```

Comment:
A space is missing after }

Reply:
Correct, I disabled auto format and missed that checking.